nautobot/core/jobs/__init__.py

```diff
@@ -282,7 +283,7 @@ def run(self, *, content_type, csv_data=None, csv_file=None, roll_back_if_error=
         if not csv_data and not csv_file:
             raise RunJobTaskFailed("Either csv_data or csv_file must be provided")
         if csv_file:
-            csv_bytes = csv_file
+            csv_bytes = codecs.EncodedFile(csv_file, "utf-8", "utf-8-sig")
```

Comment:
utf-8 is the data encoding and utf-8-sig is the file encoding. "Bytes read from the original file are decoded according to file_encoding, and the result is encoded using data_encoding." From the python [doc](https://docs.python.org/3/library/codecs.html#codecs.EncodedFile)

Reply:
Maybe add as a comment in the code?

Reply:
Do we only want a  fix for `utf-8-sig` are there not other encoding we might want to consider?
or better still only accept `UTF-8` and throw an error when encoding is not in `utf-8`

Reply:
I think the current solution takes utf-8 and utf-8-sig which is more desirable than utf-8 alone. From the issue description, it seems like a lot of the users are using utf-8 with BOM, whether advertently or inadvertently. So IMO, we should support both.